tensorboard/plugins/text/text_plugin.py

```diff
@@ -230,33 +236,59 @@ def is_active(self):
     def frontend_metadata(self):
         return base_plugin.FrontendMetadata(element_name="tf-text-dashboard")
 
-    def index_impl(self):
-        mapping = self._multiplexer.PluginRunToTagToContent(
-            metadata.PLUGIN_NAME
-        )
+    def index_impl(self, experiment):
+        if self._data_provider:
+            mapping = self._data_provider.list_tensors(
+                experiment_id=experiment, plugin_name=metadata.PLUGIN_NAME,
+            )
+        else:
+            mapping = self._multiplexer.PluginRunToTagToContent(
+                metadata.PLUGIN_NAME
+            )
         return {
             run: list(tag_to_content)
             for (run, tag_to_content) in six.iteritems(mapping)
         }
 
     @wrappers.Request.application
     def tags_route(self, request):
-        index = self.index_impl()
+        experiment = plugin_util.experiment_id(request.environ)
+        index = self.index_impl(experiment)
         return http_util.Respond(request, index, "application/json")
 
-    def text_impl(self, run, tag):
+    def text_impl(self, run, tag, experiment):
+        if self._data_provider:
+            # Downsample reads to 10 text samples per time series, which
```

Comment:
I wonder if we should make this higher?  Personally 10 has always seemed pretty low for text samples (which are generally going to not get as big as images or audio).  The multiplexer DataProvider would still of course max out at 10 samples but then at least other ones wouldn't be subject to the same constraint.

Also, hardcoding this to 10 means that the `--samples_per_plugin` flag would no longer provide an escape hatch once the DataProvider path is on by default.

In fairness, I think this is a specific instance of a more general issue which is that random sampling to a fixed _k_ steps is kind of a weird approach to use in general for the data that isn't being rendered as the entire time series at once (like scalars or histograms/distributions).  The limits we have today are more about avoiding too much memory usage in the multiplexer than really about what makes sense in terms of user experience.

But it seems like in the short term it would at least be better to make this respect `--samples_per_plugin` rather than hardcoding the default value.

Reply:
Hmm; good point. I’m happy to increase it. I hadn’t considered the
(non-)effect of the `--samples_per_plugin` flag on the generic data
path. I’ll see if I can come up with a patch for that tomorrow, as it
makes #3265 technically a minor regression.

Reply:
Went ahead and upped to 100 (why not). Will make the change to respect
`--samples_per_plugin` separately, as it touches all the plugins.

Reply:
Wait, no, it’s not technically a regression, because the multiplexer
data provider ignores `downsample` entirely, so `samples_per_plugin` in
fact still behaves how it always did. :-)

I’ll fix it the right way, anyway.